views.go

```diff
@@ -197,6 +198,14 @@ func (lk *logKeeper) createTest(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	if build.S3 {
```

Comment:
Isn't the intent that if the build was stored in S3, it wouldn't have an entry in the database? This change doesn't handle that case... not sure if that's something that should be included here or in a later change.

Reply:
My thinking is that there are three stages:
1. Now. Production data is stored in the db. If we specify we want something to go to S3 (for testing) then it's duplicated there.
2. When we're writing some percentage of production data to S3. More specifically, we'll write all the data for some percentage of _builds_ to S3. Here too everything is duplicated between the db and S3.
3. When production data all goes to S3. At this point data will not be written to the db at all.

In stages 1 and 2 we'll have this flag in the build document in the db so we know where to send the rest of a build. When we transition to stage 3 we know that _everything_ we're writing goes to S3 so we won't need this flag anymore.

Reply:
Ah that makes a lot of sense. Thanks.